### PR TITLE
master: use epoch time for timestamps (closes #726)

### LINKED ISSUE #726: use time() for timestamps
`start_time` and `run_time` should be epoch and not `localtime()` to avoid timezone, DST, and leap second issues. directories and filenames can remain in local time.

### PR DESCRIPTION
**Pull request by cjbe**
Use time.time() for experiment run_time and start_time - the results directory names are still based on local time.

### CODE REVIEW

File: artiq/master/worker_impl.py [hunk 1]
```diff
@@ -286,8 +287,8 @@ def main():
                     dataset_mgr.write_hdf5(f)
                     f["artiq_version"] = artiq_version
                     f["rid"] = rid
-                    f["start_time"] = int(time.mktime(start_time))
-                    f["run_time"] = int(time.mktime(run_time))
+                    f["start_time"] = int(start_time)
+                    f["run_time"] = int(run_time)
```
Comment (jordens) [inline artiq/master/worker_impl.py]: Let's make these doubles while we are at it. I see no reason to truncate it here.

**Comment by sbourdeauducq** [comment]
Can't we use epoch time everywhere?

**Comment by jordens** [comment]
Lab books are kept in local time and being able to correlate lab book entries with data is high value. Otherwise we could just use run id.

**Comment by sbourdeauducq** [comment]
Yes, having epoch time in files and local time everywhere else (logs, directories) can be reasonable for manual use. But it makes it a bit tricky to programmatically locate experiment files based on time.

**Comment by cjbe** [comment]
Using local time makes it much easier to correlate with other lab timestamps (i.e. a wristwatch!). As long as we have run_time in epoch time we can unambiguously recreate the experiment timeline.

**Comment by jordens** [comment]
@sbourdeauducq To make locating files by runid or epoch easier we should log those tuples `(runid,start_time,filename)` to a file.

**Comment by sbourdeauducq** [comment]
OK.

**Comment by jordens** [comment]
thanks. i amended the log entry and resolved the conflict.

**Pull request merged**